src/pages/Fox/components/FoxTokenHeader.tsx

```diff
@@ -0,0 +1,183 @@
+import { ArrowUpIcon } from '@chakra-ui/icons'
+import type { FlexProps, StackProps } from '@chakra-ui/react'
+import { Button, Flex, SimpleGrid, Skeleton, Stack } from '@chakra-ui/react'
+import { bnOrZero } from '@shapeshiftoss/utils'
+import { useCallback, useEffect } from 'react'
+import { FaCreditCard } from 'react-icons/fa'
+import { useHistory } from 'react-router'
+import { Amount } from 'components/Amount/Amount'
+import { RewardsIcon } from 'components/Icons/RewardsIcon'
+import { FiatRampAction } from 'components/Modals/FiatRamps/FiatRampsCommon'
+import { Text } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useModal } from 'hooks/useModal/useModal'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { marketApi } from 'state/slices/marketDataSlice/marketDataSlice'
+import { selectMarketDataByAssetIdUserCurrency } from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { useFoxPageContext } from '../hooks/useFoxPageContext'
+
+const arrowUpIcon = <ArrowUpIcon />
+const rewardsIcon = <RewardsIcon />
+const faCreditCardIcon = <FaCreditCard />
+
+const columnsProps = {
+  base: 1,
+  sm: 2,
+  md: 4,
+}
+const stackProps: StackProps = {
+  width: '100%',
+  flexDir: 'column',
+  flex: 1,
+  spacing: 0,
+}
+
+const containerProps: FlexProps = {
+  py: 4,
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  flexDir: {
+    base: 'column',
+    md: 'row',
+  },
+}
+
+export const FoxTokenHeader = () => {
+  const { assetId, selectedAssetAccountId } = useFoxPageContext()
+  const appDispatch = useAppDispatch()
+
+  const marketData = useAppSelector(state => selectMarketDataByAssetIdUserCurrency(state, assetId))
+  const send = useModal('send')
+  const fiatRamps = useModal('fiatRamps')
+  const {
+    state: { isConnected },
+    dispatch,
+  } = useWallet()
+  const history = useHistory()
+
+  const handleWalletModalOpen = useCallback(
+    () => dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true }),
+    [dispatch],
+  )
+
+  const handleSendClick = useCallback(
+    () =>
+      isConnected
+        ? send.open({ assetId, accountId: selectedAssetAccountId })
+        : handleWalletModalOpen(),
+    [selectedAssetAccountId, assetId, handleWalletModalOpen, isConnected, send],
+  )
+
+  const handleBuySellClick = useCallback(() => {
+    fiatRamps.open({
+      assetId,
+      fiatRampAction: FiatRampAction.Buy,
+      accountId: selectedAssetAccountId,
+    })
+  }, [selectedAssetAccountId, assetId, fiatRamps])
+
+  const handleStakeClick = useCallback(() => {
+    history.push('/rfox')
+  }, [history])
+
+  useEffect(() => {
+    if (marketData.price !== '0') return
```

Comment:
nitpick: can remove me since we don't `forceRefetch` - if market data already exists, won't be refetched

Reply:
Nice, done in [e05e4b2](https://github.com/shapeshift/web/pull/7882/commits/e05e4b244591c3a87f0da0fca531273c9a08a4c3)